controllers/controller_shared.go

```diff
@@ -87,9 +87,15 @@ func GetScopedMatchingInstances(ctx context.Context, k8sClient client.Client, cr
 		if !selected {
 			continue
 		}
+
+		// Readiness checks omits instances that have not completed a full reconciliation successfully.
+		// This allows speeding up reconciliations and reduces the amount of noisyerrors.
+		// A toggle to skip the readiness check allows additional testing of reconcilers, like provoking the ApplyFailed synchronization condition.
+		doReadinessCheck := instance.Annotations["grafana/skip-readiness-check"] == ""
```

Comment:
- Annotations are intended to have values;
- When working with boolean-like values, better to have explicitly compare to "true" / "false";
- Since we're talking about the operator checking for readiness, better to prefix the annotation with `grafana-operator`, not `grafana`.

So, in your scenario, it should be something like:

```go
doReadinessCheck := true

if instance.Annotations["grafana-operator/skip-readiness-check"] == "true" {
    doReadinessCheck = false
}
```